### PR TITLE
Introduce useUpdateConfig

### PR DESCRIPTION
**Pull request by Szymx95**
This PR will add `useSetHook` that returns `setConfig` function.

Set config accepts object as parameter that will be used to update config.

### CODE REVIEW

File: packages/core/src/providers/config/context.ts [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
 
 export function useConfig() {
-  return useContext(ConfigContext)
+  const { config } = useContext(ConfigContext)
+  return config
+}
+
+export function useSetConfig() {
+  const { setConfig } = useContext(ConfigContext)
+  return setConfig
```
Comment (marekkirejczyk) [inline packages/core/src/providers/config/context.ts:17]: Idea: Why not a single hook? ie
```ts
const {setConfig, ...config} = useConfig() 
```

or
```ts
const config = useConfig() 
config.setConfig()
```

or 
```ts
const {setConfig} = useConfig() 
```

Also:
`setConfig` or `updateConfig`?

**Review by marekkirejczyk — APPROVED** [review]

**Comment by noobshow** [comment]
While this pull request is approved and merged, I doubt @Szymx95 understood the suggestion posted https://github.com/EthWorks/useDApp/pull/242#discussion_r639794078 that @marekkirejczyk was proposing the `setConfig` or `updateConfig` to be destructured from the same useConfig and make it look like this
`const {config, setConfig} = useConfig()`
Instead, he added a new hook to support useConfig job... I guess that's like duplicating or not a good coding practice...

Hopefully, it can be fixed as this seems to be a minor version and many interfaces are bound to change.

**Comment by marekkirejczyk** [comment]
> While this pull request is approved and merged, I doubt @Szymx95 understood the suggestion posted [#242 (comment)](https://github.com/EthWorks/useDApp/pull/242#discussion_r639794078) that @marekkirejczyk was proposing the `setConfig` or `updateConfig` to be destructured from the same useConfig and make it look like this
> `const {config, setConfig} = useConfig()`

We went back and forth on this one with @Szymx95. Finally decided to be two separate hooks. The rationale is that one can use `config` quite often as a normal user. While `updateConfig` is reserved mostly for under-the-hood framework features. Also it feels right that useConfig returns just plain object without functions.

**Comment by noobshow** [comment]
I think that's something to agree with...

**Pull request merged**